bungeecord/src/main/java/io/github/retrooper/packetevents/processor/InternalBungeeProcessor.java

```diff
@@ -20,20 +20,48 @@
 
 import com.github.retrooper.packetevents.PacketEvents;
 import com.github.retrooper.packetevents.event.UserLoginEvent;
-import com.github.retrooper.packetevents.netty.channel.ChannelHelper;
+import com.github.retrooper.packetevents.manager.protocol.ProtocolManager;
 import com.github.retrooper.packetevents.protocol.player.User;
 import net.md_5.bungee.api.connection.ProxiedPlayer;
 import net.md_5.bungee.api.event.PostLoginEvent;
 import net.md_5.bungee.api.plugin.Listener;
 import net.md_5.bungee.event.EventHandler;
 
+import java.lang.reflect.Field;
+
 public class InternalBungeeProcessor implements Listener {
+
+    private static final Field CHANNEL_WRAPPER, CHANNEL;
+
+    static {
+        try {
+            CHANNEL_WRAPPER = Class.forName("net.md_5.bungee.UserConnection")
+                    .getDeclaredField("ch");
+            CHANNEL_WRAPPER.setAccessible(true);
+
+            CHANNEL = Class.forName("net.md_5.bungee.netty.ChannelWrapper")
+                    .getDeclaredField("ch");
+            CHANNEL.setAccessible(true);
+        } catch (ReflectiveOperationException exception) {
+            throw new RuntimeException("Error looking up channel fields", exception);
+        }
+    }
+
     // Called right after LoginSuccess is sent to the user.
     @EventHandler
     public void onPostLogin(PostLoginEvent event) {
         ProxiedPlayer player = event.getPlayer();
-        Object channel = PacketEvents.getAPI().getPlayerManager().getChannel(player);
+
+        Object channel;
+        try {
+            channel = CHANNEL.get(CHANNEL_WRAPPER.get(player));
+        } catch (ReflectiveOperationException exception) {
+            throw new RuntimeException("Error looking up channel from " + player, exception);
+        }
+
+        ProtocolManager.CHANNELS.put(player.getUniqueId(), channel);
         PacketEvents.getAPI().getInjector().setPlayer(channel, player);
```

Comment:
We could just simply implement the mapping into the setPlayer method implemented by BungeePipelineInjector.